frontend/public/src/factories/course.js

```diff
@@ -134,3 +146,183 @@ export const makeProgramEnrollment = (): ProgramEnrollment => ({
   program:     makeProgram(),
   enrollments: makeCourseRunEnrollment()
 })
+
+export const makeLearnerRecordCertificate = (): LearnerRecordCertificate => ({
+  uuid: casual.uuid,
+  link: casual.array_of_words(3).join("/")
+})
+
+export const makeLearnerRecordGrade = (): LearnerRecordGrade => ({
+  grade:         Math.ceil(Math.random() * 100) / 100,
+  letter_grade:  String.fromCharCode(casual.integer(0, 3) + 65),
+  passed:        casual.coin_flip,
+  set_by_admin:  casual.coin_flip,
+  grade_percent: Math.ceil(Math.random() * 100)
+})
+
+export const makeLearnerRecordCourse = (): LearnerRecordCourse => ({
+  title:       casual.short_description,
+  id:          genCourseId.next().value,
+  readable_id: casual.word.concat(genCoursewareId.next().value),
+  reqtype:     null,
+  grade:       casual.coin_flip ? makeLearnerRecordGrade() : null,
+  certificate: casual.coin_flip ? makeLearnerRecordCertificate() : null
+})
+
+export const makeLearnerRecordProgram = (): LearnerRecordProgram => ({
+  title:        casual.short_description,
+  readable_id:  casual.word.concat(genCoursewareId.next().value),
+  courses:      [],
+  requirements: []
+})
+
+export const makePartnerSchool = (): PartnerSchool => ({
+  id:    genPartnerSchoolId.next().value,
+  name:  casual.company_name,
+  email: casual.email
+})
+
+export const makeLearnerRecordShare = (
+  forPartnerSchool: boolean
+): LearnerRecordShare => ({
+  share_uuid:     casual.uuid,
+  created_on:     casual.date(),
+  updated_on:     casual.date(),
+  is_active:      casual.coin_flip,
+  user:           casual.integer(0, 1000),
+  program:        casual.integer(0, 1000),
+  partner_school: forPartnerSchool ? casual.integer(0, 1000) : null
+})
+
+export const makeLearnerRecordUser = (): LearnerRecordUser => ({
+  name:     casual.name,
+  email:    casual.email,
+  username: casual.username
+})
+
+export const makeRequirementCourseNode = (
+  course: LearnerRecordCourse,
+  parentNode: ProgramRequirement,
+  shouldBeCompleted: boolean
+): ProgramRequirement => {
+  const courseNode = {
+    id:   genProgramRequirementId.next().value,
+    data: {
+      node_type: "course",
+      program:   parentNode.data.program,
+      course:    course.id
+    },
+    children: []
+  }
+
+  if (shouldBeCompleted || casual.coin_flip) {
+    course.grade = makeLearnerRecordGrade()
+    course.certificate = makeLearnerRecordCertificate()
+  }
+
+  course.reqtype = parentNode.data.title
+
+  return courseNode
+}
+
+export const makeLearnerRecord = (
+  shouldBeCompleted: boolean
+): LearnerRecord => {
+  // This does a lot of things. It will generate a program with a handful of
+  // courses, some of which are required and some are electives, with the
+  // requisite requirements tree, and will generate grades for your fake learner
+  // depending on the flag.
+
+  const partnerSchools = []
+  const courses = []
+
+  for (let i = 0; i < 5; i++) {
```

Comment:
Why this is 5 here?

Reply:
I wanted more than one partner school in there and 5 seemed like a nice number, no other reason